### ISSUE TITLE
[Bug] Elite Four for Gen 8 is missing, so Leon is messing up the Elite Four -> Champion order for Gen 9

### ISSUE THREAD

**Issue by jaimefd** [pagefaultgames/pokerogue#1118]
- Normal Gen 9 elite four has Kieran as champion. (should be Nemona/Geeta)
- DLC Gen 9 elite four has Leon (yes, Leon, lol), as champion. (should be Kieran)
- Nemona/Geeta don't appear.

This is likely caused because in the code in src/battle.ts, line 326, the array for champions lists 10 options, while the elite four arrays only list 9 options.

- This happens because there's 9 Elite Forus (one for each gen, -1 because no Gen 8, +1 because Gen 9 has two)
- But we have 10 champion sets (one for each gen, +1 because Gen 9 has two)

In other words, Leon is just messing up the whole thing.

I didn't make a Pull Request on this issue because there's many ways to fix this, and I'm not sure how you want to implement it. But I'd suggest just adding an elite four for Gen 8, with Marnie, Hop, Bede and a random choice between a few Gym leaders, like it happens in the real game. Or just add that bad steel type guy. But I don't think Leon should be polluting the pools of other generations.

EDIT: Also, Kieran has no dialogue, unlike the other champions.

**Comment by CodeTappert** [comment]
Well according to https://bulbapedia.bulbagarden.net/wiki/Galar_League you fight

always Marnie, Hop, (Bede - he is officially not part so i dont know if we include it), Nessa, (Bea or Allister depending on the game), Raihan.

So it would be 5 fights even if we cut Bede (which we should...). I would also Cut Hop so the elite 4 all have a typ theme. 

So it would be Marine (Dark), Nessa (Water), Bea(Fighting)/Allistar(Ghost), Raihan (Dragon).

Whats your opinion about that?

**Comment by jaimefd** [comment]
That sounds fine, I don't know about the type theme tho, I would avoid reusing gym leaders in elite four whenever possible to add more variety and potentially more egg vouchers

**Comment by CodeTappert** [comment]
> That sounds fine, I don't know about the type theme tho, I would avoid reusing gym leaders in elite four whenever possible to add more variety and potentially more egg vouchers

Have a look at Larry and or the dragon guy from kanto (dont really know the english name...). They have seperate trainer classes for their second apparence. Same could be done for the gym leaders of galar
